src/datasources/perf-ds/plugin.json

```diff
@@ -10,6 +10,11 @@
   "metrics": true,
   "annotations": false,
 
+  "queryOptions": {
```

Comment:
What does this do?

Reply:
It makes Grafana automatically include the two "graph query options" from the screenshot in the pull request. It pretty much just makes Grafana know that the datasource supports overriding these settings.

[torkelo's comment](https://github.com/grafana/grafana/issues/9411#issuecomment-333934429) is the closest thing I could find to "documentation" of queryOptions.

[This is the code](https://github.com/grafana/grafana/blob/3a1f52d8a2fdd1fd3b9bdbee05a48925bf8580f1/public/app/features/panel/partials/metrics_tab.html#L36-L63) that handles the user interaction / presentation.

'maxDataPoints' isn't really necessary, I suppose, but I included it since it's used by 'maxrows' in the API calls sent to OpenNMS.

Reply:
Thanks for the reference, wasn't aware of that one.